### PR TITLE
TS-4810: Use the correct LuaJIT LDFLAGS

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by zwoop** [comment]
So, this might look a little odd with removing the flags, but, first, because of the wrong symbols, they weren't set before. And secondly, I tested this with ASAN, and it compiles and runs fine. I think we only have to worry about the sanitizer stuff when actually building the LuaJIT library (not the things using it).

**Comment by zwoop** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/683/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/579/ for details.

File: configure.ac [hunk 1]
```diff
@@ -1808,15 +1806,15 @@ iocore_include_dirs="\
 # of the generic flags, plus any Lua-specific flags so that we
 # can strip the coverage flags from Lua while keeping them by
 # default everywhere else.
-TS_ADDTO(LUA_CFLAGS, [$CFLAGS $lua_cflags])
-AC_SUBST([LUA_CFLAGS])
+TS_ADDTO(LUAJIT_CFLAGS, [$CFLAGS $lua_cflags])
```
Comment (jpeach) [inline configure.ac]: You should probably rename ``lua_cflags`` to ``luajit_cflags`` as well.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/686/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/582/ for details.

**Comment by zwoop** [comment]
I made that suggested change, going to land this unless there are objections.

**Comment by jpeach** [comment]
👍

**Pull request merged**